### PR TITLE
exported enums to seperate file

### PR DESCRIPTION
**Pull request by justinliuc**
Check the imports/exports for each folder, I'm not sure how interfaces are supposed to interact with classes etc.

Moved enums to separate file and used it as a module in the only file that actually needs access to the face-values of cards at the moment.

### CODE REVIEW

File: fish-be/src/main/types/game/card.d.ts [hunk 1]
```diff
@@ -1,6 +1,7 @@
-import { Suit, Value } from '../../game/card';
+import { Suit, Value } from '../../game/face';
 
 export interface ICard {
-  suit: Suit;
-  value: Value;
+  _suit: Suit;
+  _value: Value;
+  matches(card1: ICard, card2: ICard): boolean;
```
Comment (bliu23) [inline fish-be/src/main/types/game/card.d.ts:6]: Is this interface correct?

File: fish-be/src/main/game/player.ts [hunk 2]
```diff
@@ -28,7 +29,13 @@ export class Player {
 
   // todo: maybe needs some comparison operator
   hasCard(card: Card): boolean {
-    return true;
+    for (const c of this._hand) {
```
[the remark this answers is not in the document]
Comment (justinliuc) [reply fish-be/src/main/game/player.ts:32]: Getting around to writing tests, but haven't gotten to the player file yet

File: fish-be/src/main/types/game/card.d.ts (hunk 1 above)
Comment (justinliuc) [reply fish-be/src/main/types/game/card.d.ts:6]: No, updated locally already

File: fish-be/src/main/game/deck.ts [hunk 3]
```diff
@@ -1,2 +1,50 @@
...
+  private shuffle(): void {
+    let currentIndex = this._cards.length,
+      temporaryValue,
+      randomIndex;
+
+    // While there remain elements to shuffle...
+    while (0 !== currentIndex) {
+      // Pick a remaining element...
+      randomIndex = Math.floor(Math.random() * currentIndex);
+      currentIndex -= 1;
```
[the remark this answers is not in the document]
Comment (justinliuc) [reply fish-be/src/main/game/deck.ts:32]: I just ripped it from : https://stackoverflow.com/questions/2450954/how-to-randomize-shuffle-a-javascript-array

File: fish-be/src/tests/game/card.test.ts [hunk 4]
```diff
@@ -1,14 +1,33 @@
...
+    expect(card1.value).to.equal(value);
+  });
+
+  it('should compare card equality', () => {
+    const card1 = new Card(Suit.Club, Value.Ace);
+    const card2 = new Card(Suit.Club, Value.Ace);
+    expect(card1.matches(card2)).to.equal(true);
+  });
+
+  it('should compare card suit inequality', () => {
```
[the remark this answers is not in the document]
Comment (justinliuc) [reply fish-be/src/tests/game/card.test.ts:22]: ok

File: fish-be/src/main/types/game/card.d.ts [hunk 5]
```diff
@@ -1,6 +1,7 @@
-import { Suit, Value } from '../../game/card';
+import { Suit, Value } from '../../game/face';
 
 export interface ICard {
-  suit: Suit;
-  value: Value;
+  _suit: Suit;
+  _value: Value;
```
[the remark this answers is not in the document]
Comment (justinliuc) [reply fish-be/src/main/types/game/card.d.ts:5]: Ill look into changing it

File: fish-be/src/main/game/deck.ts [hunk 6]
```diff
@@ -1,2 +1,50 @@
...
+import { Card } from './card';
+import { IDeck } from '../types/game/deck';
+
 // this file should handle deck logic (and should be placed in the appropriate folder)
-export class Deck {}
+export class Deck implements IDeck {
+  private _cards: Card[] = [];
+
+  constructor() {
+    for (const suit in Suit) {
```
[the remark this answers is not in the document]
Comment (justinliuc) [reply fish-be/src/main/game/deck.ts:11]: src/main/game/deck.ts:11:24 - error TS2488: Type 'typeof Suit' must have a '[Symbol.iterator]()' method that returns an iterator.

11     for (const suit of Suit) {

**Pull request merged**